### PR TITLE
KAFKA-17746 Replace JavaConverters with CollectionConverters

### PR DESCRIPTION
**Pull request by Linsiyuan9**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java [hunk 1]
```diff
@@ -1635,7 +1636,7 @@ private <K, V> KafkaShareConsumer<K, V> createShareConsumer(Deserializer<K> keyD
     private void warmup() throws InterruptedException, ExecutionException, TimeoutException {
         createTopic(warmupTp.topic());
         TestUtils.waitForCondition(() ->
-                        !scala.collection.JavaConverters.seqAsJavaList(cluster.brokers().get(0).metadataCache().getAliveBrokerNodes(new ListenerName("EXTERNAL"))).isEmpty(),
+                        !CollectionConverters.asJava(cluster.brokers().get(0).metadataCache().getAliveBrokerNodes(new ListenerName("EXTERNAL"))).isEmpty(),
```
Comment (chia7712) [inline core/src/test/java/kafka/test/api/ShareConsumerTest.java]: We can use `isEmpty` directly even though it is scala code, right?

**Review by chia7712 — COMMENTED** [review]
@Linsiyuan9 thanks for this patch. There's just one small comment left. Please take a look

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java (hunk 1 above)
Comment (Linsiyuan9) [reply core/src/test/java/kafka/test/api/ShareConsumerTest.java]: hi  @chia7712 , I have recommitd the code, please have a look😊

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java [hunk 2]
```diff
@@ -54,15 +54,14 @@ public class SaslClientsWithInvalidCredentialsTest extends AbstractSaslTest {
     public static final int NUM_PARTITIONS = 1;
     public static final int BROKER_COUNT = 1;
     public static final String KAFKA_CLIENT_SASL_MECHANISM = "SCRAM-SHA-256";
-    private static final Seq<String> KAFKA_SERVER_SASL_MECHANISMS = JavaConverters.asScalaIteratorConverter(Collections.singletonList(KAFKA_CLIENT_SASL_MECHANISM).iterator()).asScala().toSeq();
+    private static final Seq<String> KAFKA_SERVER_SASL_MECHANISMS =  CollectionConverters.asScala(Collections.singletonList(KAFKA_CLIENT_SASL_MECHANISM).iterator()).toSeq();
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java]: `CollectionConverters.asScala(Collections.singletonList(KAFKA_CLIENT_SASL_MECHANISM));`

File: tools/src/test/java/org/apache/kafka/tools/other/ReplicationQuotasTestRig.java [hunk 3]
```diff
@@ -477,8 +477,7 @@ String path() {
         }
     }
 
-    @SuppressWarnings({"deprecation"})
     private static <T> Seq<T> seq(Collection<T> seq) {
-        return JavaConverters.asScalaIteratorConverter(seq.iterator()).asScala().toSeq();
+        return CollectionConverters.asScala(seq.iterator()).toSeq();
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/other/ReplicationQuotasTestRig.java]: `CollectionConverters.asScala(seq).toSeq();`

File: core/src/main/java/kafka/server/builders/LogManagerBuilder.java [hunk 4]
```diff
@@ -168,9 +168,8 @@ public LogManager build() {
         if (scheduler == null) throw new RuntimeException("you must set scheduler");
         if (brokerTopicStats == null) throw new RuntimeException("you must set brokerTopicStats");
         if (logDirFailureChannel == null) throw new RuntimeException("you must set logDirFailureChannel");
-
-        return new LogManager(JavaConverters.asScalaIteratorConverter(logDirs.iterator()).asScala().toSeq(),
-                              JavaConverters.asScalaIteratorConverter(initialOfflineDirs.iterator()).asScala().toSeq(),
+        return new LogManager(CollectionConverters.asScala(logDirs.iterator()).toSeq(),
```
Comment (chia7712) [inline core/src/main/java/kafka/server/builders/LogManagerBuilder.java]: `CollectionConverters.asScala(logDirs).toSeq(),`

File: core/src/main/java/kafka/server/builders/LogManagerBuilder.java [hunk 5]
```diff
@@ -168,9 +168,8 @@ public LogManager build() {
...
+                              CollectionConverters.asScala(initialOfflineDirs.iterator()).toSeq(),
```
Comment (chia7712) [inline core/src/main/java/kafka/server/builders/LogManagerBuilder.java]: `CollectionConverters.asScala(initialOfflineDirs).toSeq()`

**Review by chia7712 — COMMENTED** [review]
@Linsiyuan9 thanks for updates.

**Comment by Linsiyuan9** [comment]
There seems to be a wrong unit test, let me see

**Comment by chia7712** [comment]
the failed test is traced by #17366

**Pull request merged**